src/Type/Row.purs

```diff
@@ -0,0 +1,27 @@
+module Type.Row
+  ( class RowLacks
+  ) where
+
+-- Must not be exported
+foreign import data RowLacking :: Symbol -> Symbol -> Type
+
+class RowLacksF (key :: Symbol)
```

Comment:
This constraint could be summarized as "_if_ `row` contains `key`, it must occur with type `typ` in the leftmost position", yes?